tasks/main.yml

```diff
@@ -93,6 +92,9 @@
 
     - name: Create and push CIB
       include_tasks: shell_{{ ha_cluster_pacemaker_shell }}/create-and-push-cib.yml  # yamllint disable-line rule:line-length
+      # CIB changes should be done only on one of cluster nodes to avoid corruption and
+      # inconsistency of resulting cibadmin patch file.
+      run_once: true
```

Comment:
@tomjelinek This looks like a good catch to use `run_once`. Can you confirm this logic amendment? I can see use of `run_once` in many other Ansible Tasks so this change makes sense to me?

Reply:
When the role is using pcs, all CIB changes are done to a temporary file, and there is no harm in doing so on all nodes. Once the CIB is built in the file, it is being pushed to pacemaker on one node only: `run_once: true` is set for "Push CIB diff to the cluster if it has any changes" task in `create-and-push-cib.yml`. For crmsh, pushing the CIB to pacemaker is limited to one node in the same way.

I don't see any benefit in including `create-and-push-cib.yml` on one node only, at least for pcs. CIB changes are done on only one node anyway by the mechanism described above.

Reply:
You are correct, but it also means you do them on non-primary node(s) and delete anyway, which could just be avoided with run_once.

Reply:
Given it is all automated, it does not matter if some steps are repeated/duplicated and then not used. If there is a difference here between `pcs` and `crmsh` shell's, I don't see an impact.

However logically speaking - I conversely don't see any benefit for `pcs` having the CIB file on 2..n nodes if it's only done on 1 node anyway. Seems pointless to create the file that doesn't get used. Not worth debating.

@tomjelinek can you please confirm how you want to proceed?  Leave this `run_once` when using `crmsh` PCMK Shell ?

Reply:
Fair point. Right now, I don't see a strict requirement for running pcs commands building cib on all nodes. Let's make this `run_once`, then.

Reply:
Agreed 🙂 The `run_once` for the building CIB file on node 1 only will be performed for `crmsh` PCMK Shell, and current logic for building CIB file on all nodes will be performed for `pcs` PCMK Shell. Marking this resolved.